src/zino/events.py

```diff
@@ -96,6 +97,35 @@ def get(self, device_name: str, port: Optional[PortOrIPAddress], event_class: Ty
         index = EventIndex(device_name, port, event_class)
         return self._events_by_index.get(index)
 
+    def checkout(self, event_id: int) -> Event:
+        """Checks out a copy of an event that can be freely modified without being persisted"""
+        return self[event_id].model_copy(deep=True)
+
+    def commit(self, event: Event):
+        """Commits an Event object to the state, replacing any existing event by the same id.
+
+        If the event, for some reason, does not replace an existing event, indexes are rebuilt. This entrusts the
```

Comment:
Clunky sentence. Maybe "This assumes the committer does not change the identifying index attributes of a modified event."

Reply:
Agreed. Fixed.